src/actions/photosActions.ts

```diff
@@ -33,50 +34,119 @@ const fetchPhotosetRejected = (err: string) => ({
+     Server.delete(`delete/zip/${fileUrl}`)
+      .then((response) => response.data.status )
+      .catch((error) => {
+        console.error("Error Deleting the file :", error);
+      });
+ 
+
+  
+  return function () {
+    startDownloadProcess()
+    
+      .then((response) => {
+        fileUrl = response.data.url;
+        let checkStatusInterval;
+
+        const checkStatus = () => {
+         
+          checkDownloadStatus(response.data.job_id)
+            .then((status) => {
+              if (status === "SUCCESS") {
+                clearInterval(checkStatusInterval); // Stop checking once successful
+                  
+                  getDownloadFile(fileUrl).then(response => {
+                    const downloadUrl = window.URL.createObjectURL(new Blob([response.data], { type: "application/zip" }));
+
+                    const link = document.createElement("a");
+                    link.href = downloadUrl;
+                    link.setAttribute("download", "file.zip");
+                    document.body.appendChild(link);
+                    link.click();
+                    link.remove();
+                    showNotification({
+                      message: "Download Completed",
+                      title: i18n.t<string>("toasts.downloadcomplete"),
+                      color: "teal",
+                    });
+                    deleteDownloadFile(fileUrl)
+                  })
+                .catch((error) => {
+                  console.error("Error downloading:", error);
+                });
+              }
+              else if (status === "FAILURE") {
+                showNotification({
+                  message: "Download failed",
+                  title: i18n.t<string>("toasts.downloaderror"),
+                  color: "red",
+                });
+
+                clearInterval(checkStatusInterval); // Stop checking on failure as well
+                throw new Error("Download failed.");
+              }
+            })
+            .catch((error) => {
+              console.error("Error checking download status:", error);
+            });
+        };
+        // Set up an interval to periodically check the download status
+        checkStatusInterval = setInterval(checkStatus, 3000);
+    
+      })
+
+      .catch((error) => {
+        console.error("Error:", error);
+      });
   };
 }
 
+
 export function setPhotosShared(image_hashes: string[], val_shared: boolean, target_user: SimpleUser) {
-  return function cb(dispatch: Dispatch<any>) {
+  return function (dispatch: Dispatch<any>) {
     dispatch({ type: "SET_PHOTOS_SHARED" });
     Server.post(`photosedit/share/`, {
-      image_hashes,
+      image_hashes: image_hashes,
       shared: val_shared,
       target_user_id: target_user.id,
     })
       .then(() => {
-        let notificationMessage = i18n.t("toasts.unsharephoto", {
+        let notificationMessage = i18n.t<string>("toasts.unsharephoto", {
```

Comment:
If you must to specify the type, it should be `i18n.t<string|TemplateStringsArray, TOptions, string>("...")`.
As it looks too verbose and does not provide any value I suggest to remove type from translation function completely (as it was before) and let the type be inferred.